spynnaker/pyNN/models/common/neuron_recorder.py

```diff
@@ -733,49 +830,56 @@ def set_recording(self, variable, new_state, sampling_interval=None,
             raise ConfigurationException("Variable {} is not supported".format(
                 variable))
 
-    def _get_buffered_sdram(self, vertex_slice, n_machine_time_steps):
+    def _get_buffered_sdram(
+            self, vertex_slice, n_machine_time_steps, max_rewires_per_ts):
```

Comment:
I would make max_rewires_per_ts an object variable passed in during ```__init__```.  Then additionally __don't__ include it in the list of variables, instead making it active only if the value is > 0.  This makes it similar to other things.  Because this is the first example of a variable of this type (where the number of things of this type that get recorded per timestep is variable), it doesn't make sense to generalise beyond this at this point...

Reply:
When the NeuronRecorder is initially made (in the init of AbstractPopulationVertex), you don't yet know what max_rewires_per_ts will be - a Population always has to be built before a Projection to/from it, so the NeuronRecorder is built before the SynapseDynamics - you only know that these are structural when the Projection is built.  So this will have to be done slightly differently from passing it into the NeuronRecorder init... but it ought to be possible to effectively remove it from all the functions it's currently in and set it I guess.

Reply:
Ah OK, yes that is true!  But yes, a setter might be better to avoid this being passed around so much.  I guess this is the first time we have had something recorded on the synapses rather than the neurons.

Reply:
I have now reworked this so that the value isn't passed around, and also changed to not include the "rewiring" parameter in the list of (neuron) recordable variables; instead this is now treated as an "event-per-core" variable, which gives us the option to add others in the future if we come up with any.